### ISSUE TITLE
Failed to compile under windwos

### ISSUE THREAD

**Issue by etsangsplk** [tonistiigi/fifo#5]
Should check platform?

vendor\github.com\tonistiigi\fifo\fifo.go:42: undefined: unix.Mkfifo
vendor\github.com\tonistiigi\fifo\handle_nolinux.go:18: undefined: syscall.Stat_t
vendor\github.com\tonistiigi\fifo\handle_nolinux.go:19: undefined: syscall.Stat
vendor\github.com\tonistiigi\fifo\handle_nolinux.go:33: undefined: syscall.Stat_t
vendor\github.com\tonistiigi\fifo\handle_nolinux.go:34: undefined: syscall.Stat

**Comment by tonistiigi** [comment]
Yeah, windows doesn't support fifos. See https://github.com/Microsoft/go-winio equivalent support. Probably not interested in turning this lib cross platform because its purpose is to work around the limitations of unix api. Maybe if no exceptions are needed to the interface and there is a way to test for every PR.
